insecure/corruptlibp2p/spam_test.go

```diff
@@ -0,0 +1,87 @@
+package corruptlibp2p_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/network/channels"
+
+	pb "github.com/libp2p/go-libp2p-pubsub/pb"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/stretchr/testify/require"
+	corrupt "github.com/yhassanzadeh13/go-libp2p-pubsub"
+
+	"github.com/onflow/flow-go/insecure/corruptlibp2p"
+	"github.com/onflow/flow-go/insecure/internal"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/network/p2p"
+	p2ptest "github.com/onflow/flow-go/network/p2p/test"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestSpam_IHave sets up a 2 node test between a victim node and a spammer. The spammer sends a few iHAVE control messages
+// to the victim node without being subscribed to any of the same topics.
+// The test then checks that the victim node received all the messages from the spammer.
+func TestSpam_IHave(t *testing.T) {
+	const messagesToSpam = 3
+	sporkId := unittest.IdentifierFixture()
+
+	gsrSpammer := corruptlibp2p.NewGossipSubRouterSpammer(t, sporkId)
```

Comment:
For this test as we discussed earlier, the spammer and victim should be allowed to subscribe to the same topic and exchange messages over pubsub (to avoid the race condition). Hence, please add the role of the spammer as a parameter, and feed the role as a shared variable between the spammer and victim.

Reply:
84550e52b62f0f3dcde5b179a78be6d1027b5043

Reply:
3fa4920a00d2e6ab48bfc972a1196d5012796889